workflow/gccontroller/gc_controller.go

```diff
@@ -211,9 +211,24 @@ func (c *Controller) deleteWorkflow(ctx context.Context, key string) error {
 	// It should be impossible for a workflow to have been queue without a valid key.
 	namespace, name, _ := cache.SplitMetaNamespaceKey(key)
 
+	// Double check to fix retry-with-ttl bug (#12626).
+	wf, err := c.wfclientset.ArgoprojV1alpha1().Workflows(namespace).Get(ctx, name, metav1.GetOptions{})
```

Comment:
This is definitely not ideal to have another `Get`, as mentioned in https://github.com/argoproj/argo-workflows/issues/12636#issuecomment-2044237199 (and as you did as well).

I'd definitely prefer to remove it from the queue as I mentioned there.

If that's not possible, I'd suggest we actually get the Workflow from the informer cache instead -- it should have the latest. That would avoid the network request at least

Also the comment could be improved:
```suggestion
	// Double check that this workflow is still completed. If it were retried, it may be running again, per https://github.com/argoproj/argo-workflows/issues/12636
	wf, err := c.wfclientset.ArgoprojV1alpha1().Workflows(namespace).Get(ctx, name, metav1.GetOptions{})
```